libs/langchain/langchain/smith/evaluation/runner_utils.py

```diff
@@ -1299,59 +1079,57 @@ def _evaluate_strings(self, prediction, reference=None, input=None, **kwargs) ->
     """  # noqa: E501
     if kwargs:
         warnings.warn(
-            "The following arguments are deprecated and will "
-            "be removed in a future release: "
+            "The following arguments are deprecated and "
+            "will be removed in a future release: "
             f"{kwargs.keys()}.",
             DeprecationWarning,
         )
-    wrapped_model, project_name, dataset, examples = _prepare_eval_run(
-        client, dataset_name, llm_or_chain_factory, project_name
-    )
-    results = await _arun_on_examples(
+    wrapped_model, project_name, examples, configs = _prepare_run_on_dataset(
         client,
-        examples,
-        wrapped_model,
-        concurrency_level=concurrency_level,
-        project_name=project_name,
-        verbose=verbose,
-        tags=tags,
-        evaluation=evaluation,
-        input_mapper=input_mapper,
-        data_type=dataset.data_type,
-    )
-    return TestResult(
-        project_name=project_name,
-        results=results,
+        dataset_name,
+        llm_or_chain_factory,
+        project_name,
+        evaluation,
+        tags,
+        input_mapper,
+        concurrency_level,
     )
+    progress_bar = progress.ProgressBar(len(examples))
 
+    async def _async_executor(
+        example: Example, config: RunnableConfig
+    ) -> Union[dict, str, LLMResult, ChatResult]:
+        res = await _arun_llm_or_chain(
+            example,
+            wrapped_model,
+            tags=tags,
+            callbacks=config["callbacks"],
+            input_mapper=input_mapper,
+        )
+        if verbose:
+            progress_bar.increment()
```

Comment:
I was actually thinking progress bar could be a callback handler? Would remove the need for this wrapper function, but not much difference, up to you